llama_index/agent/openai_agent.py

```diff
@@ -275,27 +283,38 @@ def _chat(
         self,
         message: str,
         chat_history: Optional[List[ChatMessage]] = None,
-        function_call: Union[str, dict] = "auto",
+        tool_choice: Union[str, dict] = "auto",
         mode: ChatResponseMode = ChatResponseMode.WAIT,
     ) -> AGENT_CHAT_RESPONSE_TYPE:
-        tools, functions = self.init_chat(message, chat_history)
+        tools, openai_tools = self.init_chat(message, chat_history)
         n_function_calls = 0
 
         # Loop until no more function calls or max_function_calls is reached
-        current_func = function_call
+        current_tool_choice = tool_choice
         while True:
-            llm_chat_kwargs = self._get_llm_chat_kwargs(functions, current_func)
+            llm_chat_kwargs = self._get_llm_chat_kwargs(
+                openai_tools, current_tool_choice
+            )
             agent_chat_response = self._get_agent_response(mode=mode, **llm_chat_kwargs)
-            if not self._should_continue(self.latest_function_call, n_function_calls):
+            if not self._should_continue(self.latest_tool_calls, n_function_calls):
                 logger.debug("Break: should continue False")
                 break
-            assert isinstance(self.latest_function_call, dict)
-            self._call_function(tools, self.latest_function_call)
-            # change function call to the default value, if a custom function was given
-            # as an argument (none and auto are predefined by OpenAI)
-            if current_func not in ("auto", "none"):
-                current_func = "auto"
-            n_function_calls += 1
+            # iterate through all the tool calls
+            logger.info(f"continue to tools: {self.latest_tool_calls}")
+            for tool_call in self.latest_tool_calls:
+                # Some validation
+                if not isinstance(tool_call, dict):
+                    raise ValueError("Invalid tool_call object")
+
+                if tool_call["type"] != "function":
+                    raise ValueError("Invalid tool type. Unsupported by OpenAI")
+                # TODO: maybe execute this with multi-threading
+                self._call_function(tools, tool_call)
```

Comment:
ya let's do that for the future, in this PR just running them in a loop is fine.

Reply:
sg!